app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/MainActivity.kt

```diff
@@ -26,20 +29,30 @@ class MainActivity : AppCompatActivity() {
         setContentView(binding.root)
         binding.mainHeroList.autoFitColumnsForGridLayout(resources.getDimension(R.dimen.avatar_width))
     }
-    private fun initViewModel(){
+
+    private fun initViewModel() {
         viewModel = getViewModel { MainViewModel(Repository(application)) }
         adapter = AdapterList(viewModel::onResultClick)
         binding.mainHeroList.adapter = adapter
         viewModel.model.observe(this, Observer(::updateUI))
         viewModel.navigation.observe(this, Observer { event ->
             event.getContentIfNotHandled()?.let {
+                viewModel.viewItem.value?.let {
+                    viewItem = it
+                }
+                val options =
+                    ActivityOptionsCompat.makeSceneTransitionAnimation(
+                        this,
+                        viewItem,
```

Comment:
@jujoru if the value is not nullable, I will add the view value here directly, otherwise, the app could crash. I think it would be better if the lambda which provides the Result object, and the View from the adapter will connect with the View to store the view selected.